src/strategies_and_params/reportingstrategy.jl

```diff
@@ -154,28 +165,31 @@ function print_stats(io::IO, step, state)
 end
 
 """
-    ReportDFAndInfo(; k=1, i=100, io=stdout, writeinfo=false) <: ReportingStrategy
+    ReportDFAndInfo(; reporting_interval=1, info_interval=100, io=stdout, writeinfo=false) <: ReportingStrategy
 
-The default [`ReportingStrategy`](@ref). Report every `k`th step to a `DataFrame` and write
-info message to `io` every `i`th step (unless `writeinfo == false`). The flag `writeinfo` is
-useful for controlling info messages in MPI codes, e.g. by setting
+The default [`ReportingStrategy`](@ref). Report every `reporting_interval`th step to a `DataFrame` 
+and write info message to `io` every `info_interval`th reported step (unless `writeinfo == false`). The flag 
+`writeinfo` is useful for controlling info messages in MPI codes, e.g. by setting
 `writeinfo =`[`is_mpi_root()`](@ref).
 """
 @with_kw struct ReportDFAndInfo <: ReportingStrategy
-    k::Int = 1
-    i::Int = 100
+    reporting_interval::Int = 1
+    info_interval::Int = 100
     io::IO = stdout
     writeinfo::Bool = false
 end
 function report!(s::ReportDFAndInfo, step, args...)
-    step % s.k == 0 && report!(args...)
+    step % s.reporting_interval == 0 && report!(args...)
     return nothing
 end
 function report_after_step(s::ReportDFAndInfo, step, _, state)
-    if s.writeinfo && step % s.i == 0
+    if s.writeinfo && step % (s.info_interval * s.reporting_interval) == 0
```

Comment:
```suggestion
    if s.writeinfo && step % s.info_interval == 0
```
My suggestion would be to maintain the previous behaviour, i.e. where `info_interval` is independent of `reporting_interval`, but I'm happy to discuss. Either way, it should be clear from the docstring what is going to happen.

Reply:
The previous way has a problem if the two parameters are incommensurate, and info is not printed as expected

Reply:
It should not be, as `report_after_step()` is still called every time step (not included in the if condition). It is (and should be) completely independent of the `reporting_inteval`.